src/server/scripts/Spells/spell_dk.cpp

```diff
@@ -710,6 +711,29 @@ class spell_dk_obliteration : public AuraScript
     }
 };
 
+// 207200 - Permafrost
+class spell_dk_permafrost : public AuraScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_DK_FROST_SHIELD });
+    }
+
+    void HandleEffectProc(AuraEffect const* aurEff, ProcEventInfo& eventInfo)
+    {
+        _damageAmount = CalculatePct(eventInfo.GetDamageInfo()->GetDamage(), aurEff->GetAmount());
+        GetTarget()->CastSpell(GetTarget(), SPELL_DK_FROST_SHIELD, CastSpellExtraArgs(aurEff).AddSpellMod(SPELLVALUE_BASE_POINT0, _damageAmount));
+    }
+
+    void Register() override
+    {
+        OnEffectProc += AuraEffectProcFn(spell_dk_permafrost::HandleEffectProc, EFFECT_0, SPELL_AURA_DUMMY);
+    }
+
+private:
+    uint32 _damageAmount;
```

Comment:
Why is this a class member at all? Remove that

Reply:
done